StudentDataAnalysator/ViewModels/MeasuresCentralTrendViewModel.cs

```diff
@@ -82,17 +129,38 @@ private string MergeModesToOneString(List<double> results)
             return modesToString;
         }
 
-        private void GetAllStudentsResults()
+        private void FillDictionaryWithCoursesViewedData()
         {
-            foreach(Student student in StudentsList)
+            int count;
+            foreach (var student in StudentsList)
             {
-                results.Add(student.Result);
+                count = 0;
+                foreach (var log in LogsList)
+                {
+                    if (log.Description.Contains(student.Id.ToString()) && log.EventName == "Course viewed")
```

Comment:
This if statement is too long, you can extract the "Contains" check in a bool variable and then use it in the if statement.